workflow/controller/controller.go

```diff
@@ -379,8 +379,11 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 	<-ctx.Done()
 }
 
-func (wfc *WorkflowController) RunPrometheusServer(ctx context.Context, isDummy bool) {
-	go wfc.metrics.RunPrometheusServer(ctx, isDummy)
+func (wfc *WorkflowController) RunPrometheusServer(ctx context.Context, isDummy bool) *gosync.WaitGroup {
+	var wg gosync.WaitGroup
+	wg.Add(1)
+	go wfc.metrics.RunPrometheusServer(ctx, isDummy, &wg)
```

Comment:
or make this synchronous instead?

Reply:
Done in https://github.com/argoproj/argo-workflows/commit/5dbe063d15a3dc6bf66ccd5151a55eff042c0db1